### ISSUE TITLE
Specify minimal version dependencies + add action

### ISSUE THREAD

**Issue by pawelru** [insightsengineering/nestdevs-tasks#7]
### Overview of `verdepcheck`

Workflow: 

- Determines versions of packages to install according to strategy
- Dowload and installs each package
- Performs `R CMD check` with the packages that were installed
  - Builds vignettes
  - Runs examples
  - Runs tests
  - Notifies of API call being used that are not available _(package version might be too old)_

Strategies: 

- `min_isolated`: dependencies resolved independently with different PPM snapshots
- `min_cohort`: dependencies installed from 1 PPM snapshot, the most "minimum" version that was last released
- `release`: installs latest release of all packages
- `max`: install all dependecies from Github repos

### ~[tasks]:~ verdepcheck :package: 

- [x] https://github.com/insightsengineering/verdepcheck/pull/26
- [x] https://github.com/insightsengineering/verdepcheck/pull/24

### ~[tasks]:~ Github Actions:

- [x] https://github.com/insightsengineering/idr-tasks/issues/492
- [x] https://github.com/insightsengineering/idr-tasks/issues/513
- [x] https://github.com/insightsengineering/r-verdepcheck-action/pull/16
- [x] https://github.com/insightsengineering/r.pkg.template/pull/175

### [tasks]: Repos to review (For the 🪹 CoreDev team)

- Please add your name at the end of the PR.
- Use the checklist below as a guideline on how to review the PRs
- note that base packages don't need a `Config/Needs/verdepcheck` entry _(stats, utils, grDevices, methods, ...)_

#### 📑 Checklist for PR review:

(each PR has the checklist)

#### Repos _(please add your name at the end)_

- [x] CI: ✅ https://github.com/insightsengineering/citril/pull/313 @m7pr 
- [x] CI: ✅ https://github.com/insightsengineering/citril.metadata/pull/72
- [x] CI: ✅ https://github.com/insightsengineering/dunlin/pull/111 @donyunardi
- [x] CI: ✅ https://github.com/insightsengineering/formatters/pull/182 @vedhav
- [x] CI: ✅ https://github.com/insightsengineering/nestcolor/pull/59 
- [x] CI: ✅ https://github.com/insightsengineering/random.cdisc.data/pull/286
- [x] CI: ✅ https://github.com/insightsengineering/rlistings/pull/139
- [x] CI: ✅ https://github.com/insightsengineering/rtables/pull/693 @m7pr 
- [x] CI: ✅ https://github.com/insightsengineering/teal.code/pull/123 @chlebowa 
- [x] CI: ✅ https://github.com/insightsengineering/teal.data/pull/158 @chlebowa 
- [x] CI: ✅ https://github.com/insightsengineering/teal.modules.general/pull/538
    - (doesn't fail as `{tmc}` because it has `{teal.data}` in dependencies
- [x] CI: ✅ https://github.com/insightsengineering/teal.reporter/pull/194 @kartikeyakirar 
- [x] CI: ✅ https://github.com/insightsengineering/teal.transform/pull/150 @chlebowa 
- [x] CI: ✅ https://github.com/insightsengineering/teal.widgets/pull/168 @vedhav

-----

#### Repos _(only max is failing)_

- [x] CI: 🟠 https://github.com/insightsengineering/teal/pull/912
  - `max`: `{teal.logger}` repo problem
- [x] CI: 🟠 (test fails max) https://github.com/insightsengineering/teal.logger/pull/49
  - `max`: Conflict with `max` as test/example fails
- [x] CI: 🟠 https://github.com/insightsengineering/teal.slice/pull/405
  - `max`: Vignettes fail with `'con' is not a connection`

#### Repos _(failing)_

- [x] CI: 🔴 https://github.com/insightsengineering/chevron/pull/582
  - `min_cohort`, `min_isolated`: Can't install `{tern} 0.9.9013` no tag exists!
- [x] CI: 🔴 https://github.com/insightsengineering/teal.goshawk/pull/229
    - `all`: Secondary GH deps _(from teal)_
- [x] CI: 🔴 https://github.com/insightsengineering/teal.modules.clinical/pull/797
    - `all`: Secondary GH deps _(from `{teal.slice}` needs `{teal.data}`)_
- [x] CI: 🔴 https://github.com/insightsengineering/teal.osprey/pull/222
    - `all`: Secondary GH deps _(from `{teal}` needs `{teal.slice}`)_
- [x] CI: 🟠 https://github.com/insightsengineering/tern/pull/1012
  - `all`: fails to build {vdiffr} from CRAN & GH repo
- [x] CI: 🔴 https://github.com/insightsengineering/tern.rbmi/pull/57 @kartikeyakirar 
  - `max` & `min_isolated`
  - CI image using using `g++17`
  - `{rbmi}` requires `{TMB}` version that is failing with this `g++` version _(`g++12` works fine)_

(there are more repos but I have only included those which has integration tests at the moment of doing this task)

#### Repos _(PRs to be closed and superseded by section above)_:

<details>

<summary>Click to see abandoned PRs</summary>

- [x] https://github.com/insightsengineering/r.pkg.template/pull/158
- [x] https://github.com/insightsengineering/random.cdisc.data/pull/278
- [x] https://github.com/insightsengineering/nestcolor/pull/54
- [x] https://github.com/insightsengineering/citril.metadata/pull/66
- [x] https://github.com/insightsengineering/dunlin/pull/96
- [x] https://github.com/insightsengineering/chevron/pull/517
- [x] https://github.com/insightsengineering/citril/pull/250
- [x] https://github.com/insightsengineering/formatters/pull/163
- [x] https://github.com/insightsengineering/rtables/pull/650
- [x] https://github.com/insightsengineering/rlistings/pull/130
- [x] https://github.com/insightsengineering/tern/pull/957
- [x] https://github.com/insightsengineering/tern.rbmi/pull/54
- [x] https://github.com/insightsengineering/teal/pull/844
- [x] https://github.com/insightsengineering/teal.data/pull/148
- [x] https://github.com/insightsengineering/teal.code/pull/104
- [x] https://github.com/insightsengineering/teal.logger/pull/45
- [x] https://github.com/insightsengineering/teal.reporter/pull/189
- [x] https://github.com/insightsengineering/teal.slice/pull/332
- [x] https://github.com/insightsengineering/teal.transform/pull/137
- [x] https://github.com/insightsengineering/teal.widgets/pull/159
- [x] https://github.com/insightsengineering/goshawk/pull/195
- [x] https://github.com/insightsengineering/osprey/pull/118
- [x] https://github.com/insightsengineering/teal.goshawk/pull/218
- [x] https://github.com/insightsengineering/teal.modules.clinical/pull/780
- [x] https://github.com/insightsengineering/teal.modules.general/pull/527
- [x] https://github.com/insightsengineering/teal.osprey/pull/206

</details>

**Comment by averissimo** [comment]
Information on the reasons why some of the minimum versions are required:

Rcpp versions below `v1.0.0` don't compile in R (>= 4) :
* **purrr** use purrr (>= 0.3.0)
* **tidyr** tidyr (>= 0.8.3)
* **tibble** use tibble (>= 2.0.0)
* **rmarkdown** use rmarkdown (>= 1.11)
* **knitr** use knitr (>= 1.34)

Compilation errors _(header files changed)_
* **styler** use styler (>= 1.2.0)
* **stringr** use stringr (>= 1.4.1)
  * string**i** has problem compiling up to 1.7.3 with `isNull` not declared in scope, stringr from 2021 is needed
* **checkmate**  use checkmate (>= 2.1.0)
  * 2.1.0 :: Removed deprecated S macro `DOUBLE_EPS` from C source
* **testhat** use testhat (>= 3.0.4)
  * error: size of array ‘altStackMem’ is not an integral constant-expression until 3.0.3

Due to some functions not being exported
* **dplyr** use dplyr (>= 1.1.0)
  * where re-export from tidyselect
* **yaml** use yaml (>= 2.1.15)
  * adds function read_yaml write_yaml
* **r2rtf** use r2rtf (>= 0.3.2)
  * adds `render_safe` in `0.3.0`, but errors appear until `0.3.2`

**Comment by averissimo** [comment]
Current status:

#### Short term:

3 PRs to go, blocked by {rmarkdown} discussion: https://github.com/insightsengineering/teal.logger/pull/49#issuecomment-1735586602

https://github.com/insightsengineering/teal/pull/912, https://github.com/insightsengineering/teal.logger/pull/49 & https://github.com/insightsengineering/tern/pull/1012

-  _(with some strategies failing, as they're supposed to)_
- probable action: add `rmarkdown (>= 2.19),` to suggests and `rstudio/rmarkdown` to `Config/Needs/verdepcheck`
- alternative: Core dev or IDR needs to understand why `R CMD check` it's failing for some packages and not others in CI _(local docker image runs without issues in all cases)_
    - important: it's the R CMD check that is the problem not really the verdepcheck
    
#### Medium/long term _(will be resolved once all teal packages are on CRAN)_

Discussion on https://github.com/insightsengineering/verdepcheck/issues/38
